src/pybind/mgr/orchestrator/module.py

```diff
@@ -24,14 +24,25 @@
 from ._interface import OrchestratorClientMixin, DeviceLightLoc, _cli_read_command, \
     raise_if_exception, _cli_write_command, TrivialReadCompletion, OrchestratorError, \
     NoOrchestrator, OrchestratorValidationError, NFSServiceSpec, \
-    RGWSpec, InventoryFilter, InventoryHost, HostSpec, CLICommandMeta
+    RGWSpec, InventoryFilter, InventoryHost, HostSpec, CLICommandMeta, \
+    ServiceDescription
 
 def nice_delta(now, t, suffix=''):
     if t:
         return to_pretty_timedelta(now - t) + suffix
     else:
         return '-'
 
+
+def to_format(what, format):
+    if format == 'json':
+        return json.dumps(what, sort_keys=True)
+    elif format == 'json-pretty':
+        return json.dumps(what, indent=2, sort_keys=True)
+    elif format == 'yaml':
```

Comment:
```
bin/ceph orch ls --format=yaml       
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
usage: ceph [-h] [-c CEPHCONF] [-i INPUT_FILE] [-o OUTPUT_FILE] [--setuser SETUSER] [--setgroup SETGROUP] [--id CLIENT_ID] [--name CLIENT_NAME] [--cluster CLUSTER]
            [--admin-daemon ADMIN_SOCKET] [-s] [-w] [--watch-debug] [--watch-info] [--watch-sec] [--watch-warn] [--watch-error] [-W WATCH_CHANNEL] [--version] [--verbose]
            [--concise] [-f {json,json-pretty,xml,xml-pretty,plain}] [--connect-timeout CLUSTER_TIMEOUT] [--block] [--period PERIOD]
ceph: error: argument -f/--format: invalid choice: 'yaml' (choose from 'json', 'json-pretty', 'xml', 'xml-pretty', 'plain')
```
Hmm..seems guarding in cpp code.

Reply:
https://github.com/ceph/ceph/pull/34061/files#diff-b99cad6b446ed3108100f38e6b6249d8R336

you'll need to use the correct ceph binary